examples/validation-app/src/interceptors/validate-phone-num.interceptor.ts

```diff
@@ -0,0 +1,89 @@
+// Copyright IBM Corp. 2020. All Rights Reserved.
+// Node module: @loopback/example-validation-app
+// This file is licensed under the MIT License.
+// License text available at https://opensource.org/licenses/MIT
+
+import {
+  bind,
+  Interceptor,
+  InvocationContext,
+  InvocationResult,
+  Provider,
+  ValueOrPromise,
+} from '@loopback/context';
+import {CoffeeShop} from '../models';
+
+/**
+ * This class will be bound to the application as an `Interceptor` during
+ * `boot`
+ */
+@bind({tags: {key: ValidatePhoneNumInterceptor.BINDING_KEY}})
+export class ValidatePhoneNumInterceptor implements Provider<Interceptor> {
+  static readonly BINDING_KEY = `interceptors.${ValidatePhoneNumInterceptor.name}`;
+
+  /*
+  constructor() {}
+  */
+
+  /**
+   * This method is used by LoopBack context to produce an interceptor function
+   * for the binding.
+   *
+   * @returns An interceptor function
+   */
+  value() {
+    return this.intercept.bind(this);
+  }
+
+  /**
+   * The logic to intercept an invocation
```

Comment:
nitpick: could you make it more specific, like validate the area code, ...more other checking.

Reply:
added some description.